API.md

```diff
@@ -0,0 +1,319 @@
+      tagName,
+      subWizard,
+      ...eventInitDict?.detail,
+    },
+  });
+}
+
+export function newCloseWizardEvent(
+  wizard: WizardRequest,
+  eventInitDict?: CustomEventInit<Partial<WizardRequest>>
+): CloseWizardEvent {
+  return new CustomEvent<WizardRequest>('oscd-close-wizard', {
+    bubbles: true,
+    composed: true,
+    ...eventInitDict,
+    detail: wizard,
+  });
+}
+
+declare global {
+  interface ElementEventMap {
+    ['oscd-edit-wizard-request']: EditWizardRequest;
+    ['oscd-create-wizard-request']: CreateWizardRequest;
+    ['oscd-close-wizard']: WizardEvent;
+  }
+}
+```
+
+## Theming
+
+OpenSCD core sets the following CSS variables on the plugin:
+
+```css
+* {
+  --oscd-primary: var(--oscd-theme-primary, #2aa198);
+  --oscd-secondary: var(--oscd-theme-secondary, #6c71c4);
+  --oscd-error: var(--oscd-theme-error, #dc322f);
+
+  --oscd-base03: var(--oscd-theme-base03, #002b36);
+  --oscd-base02: var(--oscd-theme-base02, #073642);
+  --oscd-base01: var(--oscd-theme-base01, #586e75);
+  --oscd-base00: var(--oscd-theme-base00, #657b83);
+  --oscd-base0: var(--oscd-theme-base0, #839496);
+  --oscd-base1: var(--oscd-theme-base1, #93a1a1);
+  --oscd-base2: var(--oscd-theme-base2, #eee8d5);
+  --oscd-base3: var(--oscd-theme-base3, #fdf6e3);
+
+  --oscd-text-font: var(--oscd-theme-text-font, 'Roboto');
+  --oscd-icon-font: var(--oscd-theme-icon-font, 'Material Icons');
+}
+```
+
+It is expected that the fonts `--oscd-theme-text-font` and `--oscd-theme-icon-font` will be loaded in OpenSCD's `index.html` file. OpenSCD core does not load any fonts by itself.
+
+## Missing
+
+### Events for adding and removing plugins
+
+This is still needed if we want to enable plugins themselves to manage other plugins.
+
+### Plugin manifest, searchable plugin store
+
+This is still needed if we want to enable plugins to be installed and updated from within OpenSCD. If the above is implemented, this could be done by a plugin itself. Otherwise, this could be done by the OpenSCD distribution.
+
+A good candidate for a plugin manifest format is the Plugin type defined above, with the addition of a `kind` flag indicating whether it is a `"menu"` or `"editor"` plugin (or maybe `"both"`), and possibly a `version` property.
+
+### Plugin update notifications
+
+Given a plugin manifest with a `version` number, OpenSCD core could check for updates to the plugins and notify the user when updates are available.
+
+Within the current architecture, the plugin's own service worker is expected to handle this.
+
+### Deep linking to editor plugins
+
+Currently, the only way to open an editor plugin is by clicking on its tab in the tab bar. It would be nice to be able to deep link to a particular editor plugin, for example by clicking on a link in another plugin.
+
+This could be handled by the OpenSCD distribution, which could e.g. listen for `hashchange` events and open the appropriate editor plugin by setting the OpenSCD core's `editorIndex` property.
```

Comment:
```suggestion
This could be handled by the OpenSCD distribution, which could e.g. listen for `haschange` events and open the appropriate editor plugin by setting the OpenSCD core's `editorIndex` property.
```

Someone's got hashes in focus ;-)

More seriously do we have an approach for translations and changing of locale for open-scd-core that works? Or is this just a plugin concern?

Reply:
I don't currently see any other way to do translations (across arbitrary plugins implemented using different technology stacks) than to just hand the currently selected language code to the plugin and leave translation to the plugin author.

With English as a default language by convention, I expect most plugin authors will chose not to translate their plugins at all.

Since users (even from non-anglophone countries) have hitherto unanimously expressed no desire for localization of the app, another possibility would be to remove localization from OpenSCD core altogether. This would be uncomfortable for OMICRON (whose customers in some regions traditionally expect fully localized products), but might be easier on the community.

What do you think about this, @danyill and @JakobVogelsang ?

Reply:
I would not get rid of the `locale` property. Whether to have it in a plugin or even in the core component is an individual choice.

Reply:
Alright, let's leave it in.